src/Core/Content/Category/Subscriber/CategorySubscriber.php

```diff
@@ -38,8 +42,15 @@ public static function getSubscribedEvents(): array
     public function entityLoaded(EntityLoadedEvent $event): void
     {
         $salesChannelId = $event instanceof SalesChannelEntityLoadedEvent ? $event->getSalesChannelContext()->getSalesChannelId() : null;
+        $isSalesChannelEvent = $event instanceof SalesChannelEntityLoadedEvent;
 
         foreach ($event->getEntities() as $category) {
+            if ($isSalesChannelEvent && $category instanceof SalesChannelCategoryEntity) {
```

Comment:
The second check is not necessary as you have already confirmed that this is a SalesChannel event